pkg/asset/agent/manifests/nmstateconfig.go

```diff
@@ -57,14 +60,53 @@ func (*NMStateConfig) Name() string {
 // the asset.
 func (*NMStateConfig) Dependencies() []asset.Asset {
 	return []asset.Asset{
-		// TODO - use the asset to provide input for nmstateconfig
+		&agentconfig.AgentConfig{},
 	}
 }
 
 // Generate generates the NMStateConfig manifest.
 func (n *NMStateConfig) Generate(dependencies asset.Parents) error {
 
-	// TODO - will generate nmstateconfig from the asset used to provide input
+	agentConfig := &agentconfig.AgentConfig{}
+	dependencies.Get(agentConfig)
+
+	nmStateConfigs := []*aiv1beta1.NMStateConfig{}
+
+	if agentConfig.Config != nil {
+		for _, host := range agentConfig.Config.Spec.Hosts {
+			nmStateConfig := aiv1beta1.NMStateConfig{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "NMStateConfig",
+					APIVersion: "agent-install.openshift.io/v1beta1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      agentConfig.Config.ObjectMeta.Name,
+					Namespace: agentConfig.Config.ObjectMeta.Namespace,
+					Labels: map[string]string{
+						"infraenvs.agent-install.openshift.io": "ostest",
+					},
+				},
+				Spec: aiv1beta1.NMStateConfigSpec{
+					NetConfig: aiv1beta1.NetConfig{
+						Raw: []byte(host.NetworkConfig.Raw),
+					},
+				},
+			}
+			nmStateConfigs = append(nmStateConfigs, &nmStateConfig)
+		}
+
+		n.Config = nmStateConfigs
+		nmStateConfigData, err := k8syaml.Marshal(nmStateConfigs)
```

Comment:
I assume that k8syaml.Marshal treats the input as a list of separate yaml docs, to be concatenated using `---` separator lines, rather than as a list?

It would be nice to have a test to verify this. The current test data contains only one host, and we don't verify the output at all.

Reply:
I manually verified the file contents generated by `k8syaml.Marshal` with multiple hosts. When saved the generated  nmStateConfigData in a file, YAML validator verified that its a valid YAML although the separator `---` was not added to the file instead the contents were saved as a list.

Reply:
Savings the contents as a list needs to be fixed as it causes `create-cluster-and-infraenv.service` to fail with 
```
Jul 26 01:11:46 control-0.example.org podman[2833]: time="2022-07-26T01:11:46Z" 
level=fatal msg="Failed to register infraenv with assisted-service: 
error unmarshalling contents of /manifests/nmstateconfig.yaml: 
error unmarshaling JSON: while decoding JSON: json:
 cannot unmarshal array into Go value of type v1beta1.NMStateConfig"
```
this is because the `/etc/assisted/manifests/nmstateconfig.yaml` file contents are marshalled into a list by `k8syaml.Marshal`

Reply:
e.g. 
```
- apiVersion: agent-install.openshift.io/v1beta1
  kind: NMStateConfig
  ...................
  ..................
    - macAddress: 28:d2:44:d2:b2:1a
      name: enp3s1
- apiVersion: agent-install.openshift.io/v1beta1
  kind: NMStateConfig
  ...................
  ..................
```

Reply:
Yeah, we need something that you can `kubectl apply -f`, so

```
apiVersion:  agent-install.openshift.io/v1beta1
kind: NMStateConfig
spec:
    ...
---
apiVersion:  agent-install.openshift.io/v1beta1
kind: NMStateConfig
spec:
    ...
```

Reply:
I am fixing the content manually [here](https://github.com/openshift/installer/pull/6164/files#diff-40f56313d8b98bce54d14f601b3096dfef80c415816eaed3dc270d5eee4d3535R104-R114) as I don't see any library that can do it